devices/HumanLogger/HumanLogger.cpp

```diff
@@ -0,0 +1,463 @@
+            pImpl->bufferManager.push_back(pImpl->jointVelocitiesInterface,
+                                           timeNow,
+                                           "human_state::joint_velocities");
+        }
+    }
+
+    if (pImpl->settings.logHumanDynamics)
+    {
+        if(!pImpl->iHumanDynamics) {
+            yError() << logPrefix << "The IHumanDynamics pointer is null in the driver loop.";
+            askToStop();
+            return;
+        }
+
+        pImpl->jointTorquesInterface = pImpl->iHumanDynamics->getJointTorques();
+
+        if (pImpl->loggerType == LoggerType::MATLAB) {
+            pImpl->bufferManager.push_back(pImpl->jointTorquesInterface,
+                                           timeNow,
+                                           "human_dynamics::joint_torques");
+        }
+    }
+}
+
+// ======================
+// DeviceDriver interface
+// ======================
+
+bool HumanLogger::open(yarp::os::Searchable& config)
+{
+    std::lock_guard<std::mutex> guard(pImpl->loggerMutex);
+    if (!config.check("period")) {
+        yInfo() << logPrefix << "Using default period: " << DefaultPeriod << "s";
+    }
+
+    const double period = config.check("period", yarp::os::Value(DefaultPeriod)).asFloat64();
+    setPeriod(period);
+
+    // Load settings in the class
+    bool ok = pImpl->loadSettingsFromConfig(config);
+    if (!ok) {
+        yError() << logPrefix << "Problem in loading settings from config.";
+        return false;
+    }
+
+    return true;
+}
+
+void HumanLogger::impl::setLoggerType(std::string& str)
+{
+    if (!std::strcmp(str.c_str(), "matlab")) {
+        this->loggerType = LoggerType::MATLAB;
+    }
+}
+
+bool HumanLogger::impl::loadSettingsFromConfig(yarp::os::Searchable& config)
+{
+    // Check for logLevel parameter
+    this->loggerType = LoggerType::NONE;
+    if (!(config.check("LoggerType")
+          && (config.find("LoggerType").isString() || config.find("LoggerType").isList()))) {
+        yInfo() << logPrefix << "Using default LoggerType : MATLAB";
+        this->loggerType = LoggerType::MATLAB;
+    }
+    else if (config.check("LoggerType") && config.find("LoggerType").isList()) {
+        yarp::os::Bottle* loggerTypeList = config.find("LoggerType").asList();
+
+        for (size_t i = 0; i < loggerTypeList->size(); i++) {
+            std::string option = loggerTypeList->get(i).asString();
+
+            setLoggerType(option);
+        }
+    }
+    else if (config.check("LoggerType") && config.find("LoggerType").isString()) {
+        std::string option = config.find("LoggerType").asString();
+        setLoggerType(option);
+    }
+
+    // Display the current logger level
+    switch (this->loggerType) {
+        case LoggerType::MATLAB: {
+            yInfo() << logPrefix << "LoggerType set to MATLAB";
+            break;
+        }
+        default:
+            break;
+    }
+
+    yarp::os::Property prop;
+    prop.fromString(config.toString().c_str());
```

Comment:
Is there a specific reason why we are using the `Property` object and not the `Searchable` one?

Reply:
After F2F discussion, the `Property` was removed in https://github.com/robotology/human-dynamics-estimation/pull/298/commits/0431fa6d9bf3f61b8b0adc80182e92625b9ca2e4, and `Searchable` object is now directly used.